daemon/api.go

```diff
@@ -687,6 +688,17 @@ func searchStore(c *Command, r *http.Request, user *auth.UserState) Response {
 	case store.ErrUnauthenticated, store.ErrInvalidCredentials:
 		return Unauthorized(err.Error())
 	default:
+		if e, ok := err.(*url.Error); ok {
```

Comment:
Looking at this code makes me want to use https://github.com/pkg/errors - CC @mvo5 and @bboozzoo